scrapy/spiderloader.py

```diff
@@ -19,10 +20,21 @@ class SpiderLoader(object):
     def __init__(self, settings):
         self.spider_modules = settings.getlist('SPIDER_MODULES')
         self._spiders = {}
+        self._found = defaultdict(list)
         self._load_all_spiders()
 
     def _load_spiders(self, module):
         for spcls in iter_spider_classes(module):
+            self._found[spcls.name].append((module.__name__, spcls.__name__))
+            if spcls.name in self._spiders.keys():
+                import warnings
+                msg = ("There are several spiders with the same name {!r}:\n"
+                       "{}\n    This can cause unexpected behavior.".format(
+                            spcls.name,
+                            "\n".join(
+                                "        {1} (in {0})".format(mod, cls)
+                                for (mod, cls) in self._found[spcls.name])))
+                warnings.warn(msg, UserWarning)
```

Comment:
Updated.
This is the current output:

```
$ scrapy list
/home/paul/src/scrapy/scrapy/spiderloader.py:36: UserWarning: There are several spiders with the same name:

  ExampleSpider2 named 'othername' (in dupespider.spiders.example)
  OtherNameSpider named 'othername' (in dupespider.spiders.example2)

  ExampleSpider named 'example' (in dupespider.spiders.example)
  ExampleSpider named 'example' (in dupespider.spiders.example2)
  OtherExampleSpider named 'example' (in dupespider.spiders.example2)

  This can cause unexpected behavior.
  warnings.warn(msg, UserWarning)
example
othername
```